corrections/ensemble.py

```diff
@@ -47,29 +70,43 @@ def do_correction(self, lc):
             The status of the correction.
         """
         logger = logging.getLogger(__name__)
+        logger.info("Data Source: {}".format(lc.meta['task']['datasource']))
+
         # TODO: Remove in final version. Used to test execution time
-        fstart_time = time.time()
+        full_start = time.time()
 
-        # Calculate extra data for the target lightcurve
+        # Clean up the lightcurve by removing nans and ignoring data points with bad quality flags
+        og_time = lc.time.copy()
         lc = lc.remove_nans()
-        # NOTE: Is frange supposed to be missing the brackets and not join the percentiles ??
-        frange = np.percentile(lc.flux, 95) - np.percentile(lc.flux, 5) / np.mean(lc.flux)
+        lc_quality_mask = (lc.quality == 0)
+        # lc.time = lc.time[lc_quality_mask]
+        # lc.flux = lc.flux[lc_quality_mask]
+        # lc.flux_err = lc.flux_err[lc_quality_mask]
+
+        # Set up basic statistical parameters for the light curves. 
+        # frange is the light curve range from the 5th to the 95th percentile,
+        # drange is the relative standard deviation of the differenced light curve (to whiten the noise)
+        frange = (np.percentile(lc.flux, 95) - np.percentile(lc.flux, 5) )/ np.mean(lc.flux)
+
         drange = np.std(np.diff(lc.flux)) / np.mean(lc.flux)
-        lc.meta.update({ 'fmean' : np.max(lc.flux),
+        lc.meta.update({ 'fmean' : np.median(lc.flux),
```

Comment:
I think this information may already be in the `lc.meta['task']` dict.

Reply:
That would definitely be easier! Is there a place, maybe in the `TODOLIST` or the code I should be checking for stuff like this? I think there might be other places we do calculations when the data is already there somewhere. Derek and I have a bad habit of just writing code to get intermediate results instead of looking up if they're already available

Reply:
Well, it is actually my fault, because this is not really documented anywhere... But the `lc.meta['task']` dict will contain all the information from the `todolist` and `diagnostics` tables from the TODO-file. That should include median flux and RMS@1 hour, STD- and PTP-noise...

Reply:
But let's just merge this one in, and then we can have a look at that afterwards...